File: fix-import-errors.js
#!/usr/bin/env node;
const fs = require("fs");
const path = require("path")
//Function to fix import statement errors;
function fixImportErrors(content) {;
let fixed = content
  //Fix import statements missing semicolons;
  fixed = fixed.replace(/import\s*{\s*([^}]+)\s*}\s*from\s*[""]([^""]+)[""]\s*$/gm, "import { $1 } from \"$2\";")
  //Fix import statements with missing from keyword;
  fixed = fixed.replace(/import\s*{\s*([^}]+)\s*}\s*;\s*$/gm, "import { $1 } from \"lucide-react\";")
  //Fix malformed import statements;
  fixed = fixed.replace(/import\s*{\s*([^}]+)\s*;\s*}\s*from\s*[""]([^""]+)[""]\s*$/gm, "import { $1 } from \"$2\";")
  //Fix import statements with extra semicolons;
  fixed = fixed.replace(/import\s*{\s*([^}]+)\s*}\s*;\s*from\s*[""]([^""]+)[""]\s*$/gm, "import { $1 } from \"$2\";")
  //Fix import statements with missing closing brace;
  fixed = fixed.replace(/import\s*{\s*([^}]+)\s*from\s*[""]([^""]+)[""]\s*$/gm, "import { $1 } from \"$2\";")
  //Fix import statements with missing opening brace
  fixed = fixed.replace(/import\s*([^{][^}]+)\s*from\s*[""]([^""]+)[""]\s*$/gm, (match, imports, source) => {;
   ;
const cleanImports = imports.trim()
    return`import { ${cleanImports} } from "${source}";`
  })
  return fixed
}

//Function to process a single file
function processFile(filePath) {
  try {;
const content = fs.readFileSync(filePath, "utf8");
const fixed = fixImportErrors(content)
    if (content !== fixed) {
      fs.writeFileSync(filePath, fixed)
      console.log(`Fixed: ${filePath}`)
      return true
    }
    return false
  } catch (error) {
    console.error(`Error processing${filePath}:`, error.message)
    return false
  }
}

//Function to recursively find and process files
function processDirectory(dirPath) {;
let fixedCount = 0
  try {;
const items = fs.readdirSync(dirPath)
    for (const item of items) {;
const fullPath = path.join(dirPath, item);
const stat = fs.statSync(fullPath)
      if (stat.isDirectory()) {
        //Skip node_modules and other directories
        if (item !== "node_modules" && item !== ".git" && item !== ".next") {
          fixedCount+= processDirectory(fullPath)
        }
      } else if (item.endsWith(".tsx") || item.endsWith(".ts") || item.endsWith(".jsx") || item.endsWith(".js")) {
        if (processFile(fullPath)) {
          fixedCount++
        }
      }
    }
  } catch (error) {
    console.error(`Error processing directory${dirPath}:`, error.message)
  }
  
  return fixedCount
}

//Main execution console.log("Starting import error fixes...");
const totalFixed = processDirectory("/workspace")
console.log(`Fixed${totalFixed} files`)

File: fix_import_errors.py
#!/usr/bin/env python3
import os
import re
import glob

def fix_import_errors(file_path):
    """Fix malformed import statements"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Fix malformed imports
        fixes = [
            # Fix empty imports
            (r'import\s+from\s+[\'"][^\'"]+[\'"]', 'import { Helmet } from \'react-helmet-async\''),
            (r'import\s+{\s*}\s*from\s+[\'"][^\'"]+[\'"]', 'import { Helmet } from \'react-helmet-async\''),
            
            # Fix incomplete imports
            (r'import\s+{\s*(\w+)\s*}\s*from\s+[\'"][^\'"]+[\'"]\s*;\s*$', r'import { \1 } from \'react-helmet-async\';'),
            
            # Fix missing import names
            (r'import\s+{\s*(\w+)\s*}\s*from\s+[\'"][^\'"]+[\'"]\s*;\s*$', r'import { \1 } from \'react-helmet-async\';'),
        ]
        
        for pattern, replacement in fixes:
            content = re.sub(pattern, replacement, content, flags=re.MULTILINE)
        
        # Fix specific common issues
        if 'import  from' in content:
            content = content.replace('import  from', 'import { Helmet } from')
        
        if 'import {} from' in content:
            content = content.replace('import {} from', 'import { Helmet } from')
        
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Fixed import errors in: {file_path}")
            return True
        return False
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    # Find all TypeScript/JavaScript files
    patterns = [
        'app/**/*.tsx',
        'app/**/*.ts'
    ]
    
    files_to_process = []
    for pattern in patterns:
        files_to_process.extend(glob.glob(pattern, recursive=True))
    
    # Filter out node_modules and other directories we don't want to process
    files_to_process = [f for f in files_to_process if 'node_modules' not in f and 'dist' not in f and '.next' not in f]
    
    print(f"Found {len(files_to_process)} files to process")
    
    fixed_count = 0
    for file_path in files_to_process:
        if fix_import_errors(file_path):
            fixed_count += 1
    
    print(f"Fixed import errors in {fixed_count} files")

if __name__ == "__main__":
    main()